clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3369,6 +3369,13 @@ static void handleIntelNamedSubGroupSize(Sema &S, Decl *D,
                                                                 SizeType)) {
     S.Diag(Loc, diag::warn_attribute_type_not_supported) << AL << SizeStr;
   }
+
+  // If the [[intel::named_sub_group_size]] attribute spelling is used
+  // in earlier SYCL mode (SYCL 2017), we want to diagnose it as being
+  // an ignored attribute.
+  if (S.LangOpts.getSYCLVersion() == LangOptions::SYCL_2017)
+    S.Diag(AL.getLoc(), diag::warn_attribute_ignored) << AL;
```

Comment:
If we're ignoring the attribute, why are we adding it to the declaration?

Reply:
my mistake. We should not be adding it to the declaration. I will fix this. Thanks.